### PR TITLE
Mympirun

### PR DESCRIPTION
**Pull request by redfast00**

### CODE REVIEW

File: intro-HPC/ch_mympirun.tex [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+options, see the
+\href{https://github.com/hpcugent/vsc-mympirun/blob/master/README.md}{vsc-mympirun README}.
+
+\section{Basic usage}
+\label{sec:myrun-basic-usage}
+
+The most basic form of using \lstinline|mympirun| is \lstinline|mympirun [mympirun options] your_program [your_program options]|.
+
+For example, to run a program named \lstinline|hello_mpi| and give it a single argument (\lstinline|5|),
+we can run it with \lstinline|mympirun ./hello_mpi 5|.
```
Comment (boegel) [inline intro-HPC/ch_mympirun.tex]: Don't use `./` here, it'll confuse people. Usually people run provided software after loading a module, which doesn't require `./`.

`mympirun example 5`

File: intro-HPC/ch_mympirun.tex [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+\lstinline|mympirun| normally starts one process per \emph{core} on every node you assigned.
+So if you assigned 4 nodes with 16 cores each, \lstinline|mympirun| will
+start $4\cdot16=64$ test processes in total.
+
+\subsection{\texttt{-{}-hybrid}/\texttt{-h}}
+
+The \lstinline|--hybrid| option requires a positive number. This number specifies
+the number of processes started on each available physical \emph{node}. It will ignore
+the number of \emph{cores} per node.
```
Comment (boegel) [inline intro-HPC/ch_mympirun.tex]: `available cores`

File: intro-HPC/ch_mympirun.tex [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+\section{Controlling number of processes}
+
+There are four options you can choose from to control the number of processes
+\lstinline|mympirun| will start. In the following example, the program \lstinline|mpi_hello|
+prints a single line: \lstinline|Hello world from processor <node> ...|.
+
+\lstinline|mympirun| normally starts one process per \emph{core} on every node you assigned.
+So if you assigned 4 nodes with 16 cores each, \lstinline|mympirun| will
+start $4\cdot16=64$ test processes in total.
```
Comment (boegel) [inline intro-HPC/ch_mympirun.tex]: use 2 nodes as example, like you do below

File: intro-HPC/ch_mympirun.tex [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+The most basic form of using \lstinline|mympirun| is \lstinline|mympirun [mympirun options] your_program [your_program options]|.
+
+For example, to run a program named \lstinline|hello_mpi| and give it a single argument (\lstinline|5|),
+we can run it with \lstinline|mympirun ./hello_mpi 5|.
+
+
+\section{Controlling number of processes}
+
+There are four options you can choose from to control the number of processes
```
Comment (boegel) [inline intro-HPC/ch_mympirun.tex]: You don't mention the four options...

Explain `--hybrid` is the most commonly used one, only mention the others and and point to the `README` for more information?

File: intro-HPC/ch_mympirun.tex [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+#!/bin/bash
+echo "hello world"
+\end{code}
+
+And we run it like \lstinline|mympirun ./hello.sh --output output.txt|.
+
+To our suprise, this doesn't output to the file \lstinline|output.txt|, but to standard out!
+This is because \lstinline|mympirun| expects the program name and the arguments of the program to
+be its last arguments. Here, the \lstinline|--output output.txt| arguments are passed to
+\lstinline|./hello.sh| instead of to \lstinline|mympirun|.
```
Comment (boegel) [inline intro-HPC/ch_mympirun.tex]: Show the correct way to pass arguments to `mympirun`

**Comment by boegel** [comment]
Maybe also briefly mention the use of `--dry-run` (see https://github.com/hpcugent/vsc-mympirun/pull/139, I'll get that documented in the README too), in case people want to see what `mympirun` would actually be running.

**Comment by redfast00** [comment]
Don't merge yet please, I'll explain the module load without version.

File: intro-HPC/ch_mympirun.tex [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+For example, to run a program named \lstinline|example| and give it a single argument (\lstinline|5|),
+we can run it with \lstinline|mympirun example 5|.
+
+
+\section{Controlling number of processes}
+
+There are four options you can choose from to control the number of processes
+\lstinline|mympirun| will start. In the following example, the program \lstinline|mpi_hello|
+prints a single line: \lstinline|Hello world from processor <node> ...| (the sourcecode of
+\lstinline|mpi_hello| is \href{https://github.com/hpcugent/vsc-mympirun/blob/7bb2b70c4db4d780e72cb05014fc505c6f215b26/testscripts/mpi_helloworld.c}{available in the vsc-mympirun repository}).
```
Comment (boegel) [inline intro-HPC/ch_mympirun.tex]: just point to `master` rather than a specific commit?

File: intro-HPC/ch_mympirun.tex [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+
+\lstinline|mympirun| normally starts one process per \emph{core} on every node you assigned.
```
Comment (boegel) [inline intro-HPC/ch_mympirun.tex]: `By default, \lstinline|mympirun| starts one process ...`

File: intro-HPC/ch_mympirun.tex (hunk 6 above)
Comment (boegel) [reply intro-HPC/ch_mympirun.tex]: I understand the consequences, but I would still rather link to `master`. I think the changes of the file disappearing (or even changing) are quite small.

**Pull request merged**